src/utilities/eicrecon/eicrecon.cc

```diff
@@ -54,6 +55,8 @@ std::vector<std::string> EICRECON_DEFAULT_PLUGINS = {
 };
 
 int main(int narg, char** argv) {
+  ROOT::EnableThreadSafety();
```

Comment:
We might want to make this conditional on nthreads, but it doesn't need to be this PR.

Reply:
I don't know if we can do this early enough in EICrecon. `nthreads` is a JApplication parameter, and immediately after it is parsed, the topology is created and execution engine splits off the threads. I don't see how we can call `ROOT::EnableThreadSafety()` conditionally on `nthreads`, but still before we split those threads. Short of doing our own argv parsing.

I think the cost should be minimal in single-threaded running, since by definition we should not have any waiting due to the mutexes.

Reply:
https://github.com/eic/EICrecon/issues/1905